### ISSUE TITLE
git diff enhancement to add --no-ext-diff

### ISSUE THREAD

**Issue by AlbanAndrieu** [pre-commit/pre-commit#545]
Hi,

Is there a way when running command

```
pre-commit run --all-files
```

To use git diff --no-ext-diff instead of git diff

Maybe using an environment variable to add this external  --no-ext-diff parameter in the script https://github.com/pre-commit/pre-commit/blob/master/pre_commit/git.py ?

My issue comes from the fact that I am using meld in my ~/.gitconfig

```
[diff]
	external = /usr/local/bin/git-diff.sh
[push]
	default = simple
[merge]
	tool = meld
	renamelimit = 10000
```

less  /usr/local/bin/git-diff.sh

```
#!/bin/bash
meld "$2" "$5" > /dev/null 2>&1
```

So every-time pre-commit run --all-files run I have a popup with meld that I must close.
It is annoying...

Feel free to help me to find the right code and I might find some time to do a pull request...

Best regards,

Alban

**Comment by asottile** [comment]
Here's a PR which should solve this: https://github.com/pre-commit/pre-commit/pull/546

**Comment by asottile** [comment]
This is available in the 0.14.2 release!

**Comment by AlbanAndrieu** [comment]
This is working.
Super fix in no time...
